### PR TITLE
doc: add clarity to spec definition in README

### PR DESCRIPTION
**Pull request by MeroveeCharrue**
This is a note in the doc to prevent potential known issues when declaring endpoints to match routes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,13 +114,21 @@ At the end in the controller, you will have true json, xml for payload & integer
...
 * `GET /item/{itemId}/comment` -> `Controllers\Item\Comment#getAction`
 * `PUT /item/{itemId}/comment/{commentId}` -> `Controllers\Item\Comment#putAction`
 
-> note you can define the controller namespace (here Controllers) in the RREST\RREST constructor
+> Note: You can define the controller namespace (here Controllers) in the RREST\RREST constructor
+
+> Note: Currently, the order of API definitions matters. See this as an example :
+> ```raml
+> /item/{itemId}
+> /item/archive
```
Comment (yannickroger) [inline README.md]: I would add a working example, specify that this is a best practice and warn about the side effect if not applied.

Comment (MeroveeCharrue) [reply README.md]: Thanks, I've rephrased it to emphasis good practice.

**Comment by RETFU** [comment]
@yannickroger good for you too?

**Comment by yannickroger** [comment]
@RETFU 👍

**Review by yannickroger — APPROVED** [review]

**Pull request merged**